interface/src/app/genesets/use.js

```diff
@@ -12,354 +12,409 @@
+
+          if (scope.mygenesets === true) {
+            UserFactory.getPromise().$promise.then(function() {
+              query["creator"] = UserFactory.getUser()["id"];
+              GeneSetSearch.query(query); //query resource
             });
-        };
-
-
-        $scope.openDownloadModal = function ( versionHash ) {
-
-            var modalInstance = $modal.open({
-                templateUrl: 'genesets/download/downloadModal.tpl.html',
-                controller: ['$scope', '$modalInstance', 'CrossrefDBs', 
-                    function( $scope, $modalInstance, CrossrefDBs ) {
-                        $scope.versionHash = versionHash;
-                        $scope.crossRefDbList = ['Symbol'];
-
-                        CrossrefDBs.query( function(data) {
-                          for (var i in data.objects ) {
-                            $scope.crossRefDbList.push(data.objects[i]['name']);
-                          }
-                        });
-
-                        $scope.geneIdentifier = '';
-                        $scope.download = function () {
-                            $modalInstance.close($scope.geneIdentifier);
-                        };
-                        $scope.cancel = function () {
-                            $modalInstance.dismiss('cancel');
-                        };
-                    }
-                ],
-                resolve: {
-                }               
+          } else {
+            if ("creator" in query) {
+              delete query["creator"];
+            }
+            GeneSetSearch.query(query); //query resource
+          }
+        });
+      },
+      replace: true,
+      restrict: "E",
+      scope: {
+        limit: "=",
+        mygenesets: "="
+      },
+      templateUrl: "genesets/list/search-result-table.tpl.html"
+    };
+  })
+
+  /**
+   * Controller that handles displaying a page for each GeneSet.
+   */
+  .controller("GeneSetDetailCtrl", function(
+    $scope,
+    $stateParams,
+    $modal,
+    $state,
+    $location,
+    GeneSets,
+    UserFactory,
+    Versions
+  ) {
+    // Get Geneset
+    var gsParams = {
+      creator: $stateParams.creator,
+      slug: $stateParams.slug,
+      show_versions: true,
+      show_team: true
+    };
+
+    $scope.geneset = GeneSets.get(gsParams);
+
+    $scope.showVersion = function(chosenVersionHash) {
+      var verParams = {
+        creator: $stateParams.creator,
+        slug: $stateParams.slug,
+        version: chosenVersionHash,
+        xrids_requested: true
+      };
+      $scope.chosenVersion = Versions.get(verParams);
+      $location.search({ version: chosenVersionHash });
+    };
+    $scope.geneset.$promise.then(function(data) {
+      var initialVersion = $scope.geneset.versions[0]["ver_hash"];
+      var versionParam = $location.search().version;
+
+      if (versionParam) {
+        var matchingVersions = $scope.geneset.versions.filter(function(
```

Comment:
The search of `versionParam` doesn't have to scan the whole `$scope.geneset.versions`. It will be done when one match is found. So it can be written like this:
```
if (versionParam) {
  var found = $scope.geneset.versions.find(function(version) {
    return version.ver_hash === versionParam;
  });
  if (found) {
    initialVersion = versionParam;
  }
}
...
```
Of course, if `$scope.geneset.versions` includes only a few elements, then no big deal.